source/state_representation/include/state_representation/space/joint/JointState.hpp

```diff
@@ -584,6 +584,14 @@ class JointState : public State {
    */
   void set_state_variable(const Eigen::VectorXd& new_value, const JointStateVariable& state_variable_type);
 
+  /**
+   * @brief Print the joint state
+   * @param class_name The name of the class (one of JointState, JointPositions, JointVelocities,
+   * JointAccelerations, JointTorques)
+   * @param state_variable_type The corresponding type of variable to print
+   */
+  std::stringstream print(const std::string& class_name, const JointStateVariable& state_variable_type) const;
```

Comment:
Why are you taking the class name as input and not overloading it in the derived classes?

Reply:
I guess my idea was to have a generic helper such that I don't have to overload it and reduce code duplication. Overloading without the class name as argument would mean I'd have to repeat some code again

Reply:
Fair enough. Then maybe there is reliable way to get it from the object itself. The other possibility is to use the `StateType`. We could have a function in `State` that map the type to the string name.

Reply:
That, I could live with yeah. Good idea